src/dmd/backend/cdef.d

```diff
@@ -875,7 +875,11 @@ union eve
         {
             elem* E1;           // left child for unary & binary nodes
             elem* E2;           // right child for binary nodes
-            Symbol* Edtor;      // OPctor: destructor
+            union
+            {
+                Symbol* Edtor;  // OPctor: destructor
+                Symbol* Edecl2; // VarDeclaration being constructed
```

Comment:
When is Edecl2 valid?

Reply:
Edecl2 holds the temporary symbol being constructed. It's just used to find which `this` nodes correspond to which `ctor` node without assuming much about the tree. Nodes which share the same Edecl2 symbol are related together.